packages/Autocomplete/src/index.tsx

```diff
@@ -0,0 +1,84 @@
+import Input from "@blaze-react/input";
+import utils from '@blaze-react/utils';
+import React, { Fragment, useState } from "react";
+
+interface IAutocompleteProps {
+  data: {
+    keyValue: string;
+    filterBy: any[];
+    data: object[];
+  };
+  utils: {
+    uniqueId: (element: any) => string
+  };
+  selected: (...args: any[]) => any;
+  placeholder?: string;
+  children?: any;
+}
+const Autocomplete: React.SFC<IAutocompleteProps> = ({
+  data: { data, filterBy: keys, keyValue },
+  placeholder,
+  selected,
+  utils: {
+    uniqueId
+  },
+  children
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [showSelect, setShowSelect] = useState(false);
+
+  const handleChange = ({ value }: { value: string }) => {
+    setShowSelect(true);
+    setInputValue(value)
+    filterByValue(value);
+  };
+
+  const filterByValue = (value: string) => {
+    return data.filter(copy => {
+      let isMatch = false;
+      keys.forEach(key => {
+        const match = copy[key].toLowerCase().includes(value.toLowerCase());
+        if (match) {
+          isMatch = match;
+        }
+        if (!value) {
+          isMatch = false;
+        }
+      });
+      return isMatch;
+    });
+  }
+
+  const handleClick = (copiedData: object) => {
```

Comment:
copied data could have a more specific type or would be the keys dynamic? Also you could type the return as void

Reply:
yes, they will be dynamic